ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlRestServiceContextFactory.java

```diff
@@ -35,43 +37,61 @@
  */
 public class KsqlRestServiceContextFactory implements Factory<ServiceContext> {
   private static KsqlConfig ksqlConfig;
+  private static KsqlSecurityExtension securityExtension;
 
-  public static void configure(final KsqlConfig ksqlConfig) {
+  public static void configure(
+      final KsqlConfig ksqlConfig,
+      final KsqlSecurityExtension securityExtension
+  ) {
     KsqlRestServiceContextFactory.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig");
+    KsqlRestServiceContextFactory.securityExtension
+        = Objects.requireNonNull(securityExtension, "securityExtension");
+  }
+
+  @VisibleForTesting
+  @FunctionalInterface
+  interface ServiceContextFactory {
```

Comment:
I was confused by the naming in this file since both `defaultServiceContextFactory` and `userServiceContextFactory` are service context factories, yet only one of them is of type `ServiceContextFactory`. Can we either switch `defaultServiceContextFactory` to have type `ServiceContextFactory` or rename `ServiceContextFactory` to `UserServiceContextFactory` to avoid this confusion? I think the latter option is easier.

Same comment applies to the interface in WSQueryEndpoint as well.

Reply:
That's a good name, I will change itto `UserServiceContextFactory`.